### PR TITLE
feat: tensor coersion

### PR DESCRIPTION
**Pull request by samsja**
# Context

allow `TorchTensor` to cast from `TensorFlowTensor` and vice versa.

### CODE REVIEW

File: tests/units/typing/tensor/test_tensor_flow_tensor.py [hunk 1]
```diff
@@ -199,3 +200,25 @@ def test_set_item():
     assert tnp.allclose(t.tensor[0], tf.ones((1, 224, 224)))
     assert tnp.allclose(t.tensor[1], tf.zeros((1, 224, 224)))
     assert tnp.allclose(t.tensor[2], tf.zeros((1, 224, 224)))
+
+
+@pytest.mark.tensorflow
```
[the remark this answers is not in the document]
Comment (samsja) [reply tests/units/typing/tensor/test_tensor_flow_tensor.py:205]: sure

[the remark this answers is not in the document]
Comment (samsja) [reply tests/units/typing/tensor/test_tensor_flow_tensor.py]: done

File: tests/units/typing/tensor/test_tensor_coercion.py [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+    TensorFlowTensor._docarray_from_native(tf.zeros((3, 224, 224))),
+]
+
+
+@pytest.mark.tensorflow
+@pytest.mark.parametrize('tensor', pure_tensor_to_test + docarray_tensor_to_test)
+@pytest.mark.parametrize('tensor_cls', [NdArray, TorchTensor, TensorFlowTensor])
+def test_torch_tensor_coerse(tensor_cls, tensor):
+    t = parse_obj_as(tensor_cls, tensor)
+    assert isinstance(t, tensor_cls)
```
Comment (JoanFM) [inline tests/units/typing/tensor/test_tensor_coercion.py:26]: can we also check they are all zeroes? and that the shape is respected and so on?

**Review by JoanFM — CHANGES_REQUESTED** [review]

File: tests/units/typing/tensor/test_tensor_coercion.py (hunk 2 above)
Comment (samsja) [reply tests/units/typing/tensor/test_tensor_coercion.py:26]: sure

**Review by JoanFM — APPROVED** [review]

**Pull request merged**